includes/class-wcpdf-updraft-semaphore.php

```diff
@@ -153,48 +146,54 @@ public function lock($retries = 0) {
 	 *
-	 * @return Boolean - if it returns false, then the lock was apparently not locked by us (and the caller will most likely therefore ignore the result, whatever it is).
+	 * @return bool - if it returns false, then the lock was apparently not locked by us (and the caller will most likely therefore ignore the result, whatever it is).
 	 */
-	public function release() {
+	public function release(): bool {
 		if ( ! $this->acquired ) {
 			return false;
 		}
 
 		global $wpdb;
 		$sql = $wpdb->prepare( "UPDATE {$wpdb->options} SET option_value = '0' WHERE option_name = %s", $this->option_name );
 
-		$this->log( 'Lock option ('.$this->option_name.', '.$wpdb->options.') released', 'info' );
+		$this->log( 'Lock option (' . $this->option_name . ', ' . $wpdb->options . ') released', 'info' );
 
-		$result = (int) $wpdb->query($sql) === 1;
+		$result = (int) $wpdb->query( $sql ) === 1;
 
 		$this->acquired = false;
 
 		return $result;
 	}
 
 	/**
-	 * Cleans up the DB of any residual data. This should not be used as part of ordinary unlocking; only as part of deinstalling, or if you otherwise know that the lock will not be used again. If calling this, it's redundant to first unlock (and a no-op to attempt to do so afterwards).
+	 * Cleans up the DB of any residual data. This should not be used as part of ordinary unlocking; only as part of deinstalling, or if you otherwise know that the lock will not be used again.
+	 * If calling this, it's redundant to first unlock (and a no-op to attempt to do so afterwards).
+	 * 
+	 * @return void
 	 */
-	public function delete() {
+	public function delete(): void {
 		$this->acquired = false;
 
 		global $wpdb;
 		$wpdb->query( $wpdb->prepare( "DELETE FROM {$wpdb->options} WHERE option_name = %s", $this->option_name ) );
 
-		$this->log( 'Lock option ('.$this->option_name.', '.$wpdb->options.') was deleted from the database' );
+		$this->log( 'Lock option (' . $this->option_name . ', ' . $wpdb->options . ') was deleted from the database' );
 	}
 
 	/**
 	 * Captures and logs any given messages
 	 *
-	 * @param String $message - the error message
-	 * @param String $level	  - the message level (debug, notice, info, warning, error)
-	 * @param Array  $context - Optional. Additional information for log handlers.
+	 * @param string $message - the error message
+	 * @param string $level   - the message level (debug, notice, info, warning, error)
+	 * @param array  $context - Optional. Additional information for log handlers.
+	 * 
+	 * @return void
 	 */
-	public function log( $message, $level = 'info', $context = array() ) {
-		$context = ! empty( $context ) ? $context : $this->context;
+	public function log( string $message, string $level = 'info', array $context = array() ): void {
+		$context      = ! empty( $context ) ? $context : $this->context;
+		$logs_enabled = isset( WPO_WCPDF()->settings->debug_settings['semaphore_logs'] );
```

Comment:
I believe it depends on our settings, as we have an option that enables this type of logging. I prefer having a single check here rather than multiple checks throughout the plugin.